language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddToConfigTomlCodeAction.java

```diff
@@ -0,0 +1,288 @@
+
+    private static final String CONFIG_TOML = "Config.toml";
+
+    @Override
+    public List<SyntaxKind> getSyntaxKinds() {
+        return List.of(SyntaxKind.MODULE_VAR_DECL);
+    }
+
+    @Override
+    public boolean validate(CodeActionContext context, RangeBasedPositionDetails positionDetails) {
+        return context.currentModule().isPresent()
+                && hasConfigurableQualifier((ModuleVariableDeclarationNode) positionDetails.matchedCodeActionNode());
+    }
+
+    @Override
+    public List<CodeAction> getCodeActions(CodeActionContext context, RangeBasedPositionDetails posDetails) {
+        Module module = context.currentModule().get();
+        Project project = module.project();
+        if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
+            return Collections.emptyList();
+        }
+        String orgName = project.currentPackage().packageOrg().value();
+        SemanticModel semanticModel = context.currentSemanticModel().get();
+        Types types = semanticModel.types();
+        TypeBuilder builder = types.builder();
+        TypeSymbol basicType = builder.UNION_TYPE.withMemberTypes(types.INT, types.STRING, types.BOOLEAN,
+                types.BYTE, types.FLOAT, types.DECIMAL, types.XML).build();
+        TypeSymbol anyDataOrJsonType = builder.UNION_TYPE.withMemberTypes(types.ANYDATA, types.JSON).build();
+
+        Optional<Symbol> confSymbol = semanticModel.symbol(posDetails.matchedCodeActionNode());
+        if (confSymbol.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ConfigurableFinder.ModuleConfigDetails moduleConfigDetails = ConfigurableFinder.getModuleConfigDetails(context);
+        Map<String, ConfigurableFinder.ConfigVariable> configVarMap = moduleConfigDetails.configVariables();
+        VariableSymbol variableSymbol = (VariableSymbol) confSymbol.get();
+        ConfigurableFinder.ConfigVariable selectedConf = configVarMap.get(variableSymbol.getName().get());
+
+        Path configTomlPath = context.workspace().projectRoot(context.filePath()).resolve(CONFIG_TOML);
+        if (!configTomlPath.toFile().exists()) {
+            if (configVarMap.isEmpty()) {
+                return Collections.emptyList();
+            }
+            return withoutConfigTomlCodeActions(basicType, anyDataOrJsonType, moduleConfigDetails,
+                    selectedConf, configTomlPath);
+        }
+
+        Optional<Toml> toml = readToml(configTomlPath);
+        if (toml.isEmpty()) {
+            return Collections.emptyList();
+        }
+        int lastNodeLine = module.isDefaultModule() ? rootModuleConfigDiff(toml.get(),
+                orgName, module.moduleName().toString(), configVarMap)
+                : nonRootModuleConfigDiff(toml.get(), orgName, module.moduleName(), configVarMap);
+
+        if (configVarMap.isEmpty() || !configVarMap.containsKey(selectedConf.name())) {
+            return Collections.emptyList();
+        }
```

Comment:
Can't we move this block up, to before computing the `lastNodeLine` if `lastNodeLine` is not used for this block? 

```suggestion
        if (toml.isEmpty() || configVarMap.isEmpty() || !configVarMap.containsKey(selectedConf.name())) {
            return Collections.emptyList();
        }
```

Reply:
Moved the `configVarMap` being empty